Combine backtrace print into one string to avoid interleaving.

pritamdamania87 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#56961 Combine backtrace print into one string to avoid interleaving.**

As described in https://github.com/pytorch/pytorch/issues/56583, the
backtrace amongst several processes was garbled.
https://github.com/pytorch/pytorch/pull/56198 would've alleviated this to some
extent, but this PR combines all the logging into just one string to reduce
interleaving further.

Differential Revision: [D28013191](https://our.internmc.facebook.com/intern/diff/D28013191/)

pritamdamania committed:
Combine backtrace print into one string to avoid interleaving.

As described in https://github.com/pytorch/pytorch/issues/56583, the
backtrace amongst several processes was garbled.
https://github.com/pytorch/pytorch/pull/56198 would've alleviated this to some
extent, but this PR combines all the logging into just one string to reduce
interleaving further.

Differential Revision: [D28013191](https://our.internmc.facebook.com/intern/diff/D28013191/)

[ghstack-poisoned]

pritamdamania committed:
Update on "Combine backtrace print into one string to avoid interleaving."


As described in https://github.com/pytorch/pytorch/issues/56583, the
backtrace amongst several processes was garbled.
https://github.com/pytorch/pytorch/pull/56198 would've alleviated this to some
extent, but this PR combines all the logging into just one string to reduce
interleaving further.

Differential Revision: [D28013191](https://our.internmc.facebook.com/intern/diff/D28013191/)

[ghstack-poisoned]

pritamdamania committed:
Update on "Combine backtrace print into one string to avoid interleaving."


As described in https://github.com/pytorch/pytorch/issues/56583, the
backtrace amongst several processes was garbled.
https://github.com/pytorch/pytorch/pull/56198 would've alleviated this to some
extent, but this PR combines all the logging into just one string to reduce
interleaving further.

Differential Revision: [D28013191](https://our.internmc.facebook.com/intern/diff/D28013191/)

[ghstack-poisoned]

cbalioglu reviewed (approved):
Nice, didn't know that we could use libfmt in PyTorch. Definitely one of my favorite C++ libs.

pritamdamania committed:
Update on "Combine backtrace print into one string to avoid interleaving."


As described in https://github.com/pytorch/pytorch/issues/56583, the
backtrace amongst several processes was garbled.
https://github.com/pytorch/pytorch/pull/56198 would've alleviated this to some
extent, but this PR combines all the logging into just one string to reduce
interleaving further.

Differential Revision: [D28013191](https://our.internmc.facebook.com/intern/diff/D28013191/)

[ghstack-poisoned]

pritamdamania committed:
Update on "Combine backtrace print into one string to avoid interleaving."


As described in https://github.com/pytorch/pytorch/issues/56583, the
backtrace amongst several processes was garbled.
https://github.com/pytorch/pytorch/pull/56198 would've alleviated this to some
extent, but this PR combines all the logging into just one string to reduce
interleaving further.

Differential Revision: [D28013191](https://our.internmc.facebook.com/intern/diff/D28013191/)

[ghstack-poisoned]

cbalioglu reviewed (approved):
LGTM

pritamdamania committed:
Update on "Combine backtrace print into one string to avoid interleaving."


As described in https://github.com/pytorch/pytorch/issues/56583, the
backtrace amongst several processes was garbled.
https://github.com/pytorch/pytorch/pull/56198 would've alleviated this to some
extent, but this PR combines all the logging into just one string to reduce
interleaving further.

Differential Revision: [D28013191](https://our.internmc.facebook.com/intern/diff/D28013191/)

[ghstack-poisoned]

pritamdamania committed:
Update on "Combine backtrace print into one string to avoid interleaving."


As described in https://github.com/pytorch/pytorch/issues/56583, the
backtrace amongst several processes was garbled.
https://github.com/pytorch/pytorch/pull/56198 would've alleviated this to some
extent, but this PR combines all the logging into just one string to reduce
interleaving further.

Differential Revision: [D28013191](https://our.internmc.facebook.com/intern/diff/D28013191/)

[ghstack-poisoned]

pritamdamania committed:
Update on "Combine backtrace print into one string to avoid interleaving."


As described in https://github.com/pytorch/pytorch/issues/56583, the
backtrace amongst several processes was garbled.
https://github.com/pytorch/pytorch/pull/56198 would've alleviated this to some
extent, but this PR combines all the logging into just one string to reduce
interleaving further.

Differential Revision: [D28013191](https://our.internmc.facebook.com/intern/diff/D28013191/)

[ghstack-poisoned]
